### DISCUSSION TITLE
Auto-generated properties in CRUD operation

### DISCUSSION THREAD

**Discussion by louiskhchan** [mockoon/mockoon#1256] [Ideas]
Hi,

I have a question about mocking the server response for POST requests in Mockoon. On our server, when we use POST to create an object, the server adds some additional properties (like item ID, creation date, etc.) to that object and returns it as the response.

For example, if we POST the following JSON object to the server:
```json
{
  "task": "Buy groceries" 
}
```

The server may respond with something like this:

```json
{
  "id": 123,
  "task": "Buy groceries",
  "creationDate": "2023-02-25T14:25:29.590Z"
}
```

However, Mockoon does not support this behavior because the CRUD route only reads the JSON body from a POST request and saves it in the data bucket as is. Are there any plans to add this feature to Mockoon? For instance, can you enable us to create a template for the new object in the data bucket and reference the incoming JSON object in that template?

**Comment by 255kb** [comment]
Hi, 
This was already requested once or twice I think, but there is no plan to add this, or an opened issue yet. 
It would make sense to add a way to provide an object that could be used to populate the missing fields on POST or PUT operations. 
I'm just wondering, how it would look UI wise.

**Reply by louiskhchan** [reply]
Hmm... Maybe under the 'CRUD operations' response add a tab next to Data and Headers to allow specifying a record creation template?

**Reply by 255kb** [reply]
You are right, that could be a good place. But maybe, as we already have an "id property" field, and a lot of empty space below, we could add a small editor.
